nixos/tests/phpfpm.nix

```diff
@@ -0,0 +1,71 @@
+import ./make-test-python.nix ({ pkgs, lib, ... }: {
+  name = "phpfpm";
+  meta.maintainers = [ lib.maintainers.aanderse ];
+
+  machine =
+    { config, ... }:
+    { services.httpd.enable = true;
+      services.httpd.adminAddr = "admin@example.com";
+      services.httpd.extraModules = [ "proxy_fcgi" ];
+      services.httpd.extraConfig = ''
+        Alias /site1 /var/www/site1
+        <Directory /var/www/site1>
+          <FilesMatch "\.php$">
+            <If "-f %{REQUEST_FILENAME}">
+              SetHandler "proxy:unix:${config.services.phpfpm.pools.site1.socket}|fcgi://localhost/"
+            </If>
+          </FilesMatch>
+          DirectoryIndex index.php
+          Require all granted
+        </Directory>
+
+        Alias /site2 /var/www/site2
+        <Directory /var/www/site2>
+          <FilesMatch "\.php$">
+            <If "-f %{REQUEST_FILENAME}">
+              SetHandler "proxy:fcgi://127.0.0.1:9000"
+            </If>
+          </FilesMatch>
+          DirectoryIndex index.php
+          Require all granted
+        </Directory>
+      '';
+      services.phpfpm.pools = {
+        site1 = {
+          user = config.services.httpd.user;
+          group = config.services.httpd.group;
+          settings = {
+            "listen.owner" = config.services.httpd.user;
+            "listen.group" = config.services.httpd.group;
+            "pm" = "static";
+            "pm.max_children" = 2;
+          };
+        };
+        site2 = {
+          user = config.services.httpd.user;
+          group = config.services.httpd.group;
+          settings = {
```

Comment:
we might want to run that sites phpfpm pool as another user.

Reply:
In general I agree but for a simple test I'm less motivated. If you feel like this serves for good documentation I'll gladly accept a suggestion or commit.

Reply:
I can include add this to the test, once I fully grasped the rest :-)